### DISCUSSION TITLE
ErrnoException: Cannot read from file

### DISCUSSION THREAD

**Discussion by Sallery-X** [juicedata/juicefs#366] [General]
![image](https://user-images.githubusercontent.com/45587738/116356215-931bcf00-a82d-11eb-8ec9-6fb3492faf70.png)

**Comment by Sallery-X** [comment]
strerror: Input/output error

**Comment by davies** [comment]
Can you check the syslog?

IOError usually means it failed to talk to Redis or object storage, there should be detailed logging in syslog.

**Comment by Sallery-X** [comment]
where could I find the syslog?

**Comment by SandyXSD** [comment]
Please take a look at [this doc](https://github.com/juicedata/juicefs/blob/main/docs/en/fault_diagnosis_and_analysis.md)

**Comment by Sallery-X** [comment]
> Can you check the syslog?
> 
> IOError usually means it failed to talk to Redis or object storage, there should be detailed logging in syslog.

![image](https://user-images.githubusercontent.com/45587738/116401281-a2b30c00-a85d-11eb-9982-b474ab302000.png)

**Comment by davies** [comment]
can you post all the logs, not just the ERROR?

**Comment by Sallery-X** [comment]
> can you post all the logs, not just the ERROR?
Apr 29 11:01:50 A06-R08-I135-39-5000418 juicefs[8812]: juicefs[8812] <ERROR>: read file 443: input/output error
Apr 29 11:01:50 A06-R08-I135-39-5000418 juicefs[8812]: juicefs[8812] <INFO>: slow operation: read (443,4096,0): input/output error (0) <20.815054>
Apr 29 11:02:32 A06-R08-I135-39-5000418 juicefs[8812]: juicefs[8812] <WARNING>: fail to read chunkid 2476 (off:0, size:1, clen: 1): get chunks/0/2/2476_0_1: open /var/jfs/clickhouse/chunks/0/2/2476_0_1: no such file or directory
Apr 29 11:02:33 A06-R08-I135-39-5000418 juicefs[8812]: juicefs[8812] <WARNING>: fail to read chunkid 2476 (off:0, size:1, clen: 1): get chunks/0/2/2476_0_1: open /var/jfs/clickhouse/chunks/0/2/2476_0_1: no such file or directory
Apr 29 11:02:34 A06-R08-I135-39-5000418 juicefs[8812]: juicefs[8812] <WARNING>: fail to read chunkid 2476 (off:0, size:1, clen: 1): get chunks/0/2/2476_0_1: open /var/jfs/clickhouse/chunks/0/2/2476_0_1: no such file or directory
Apr 29 11:02:36 A06-R08-I135-39-5000418 juicefs[8812]: juicefs[8812] <WARNING>: fail to read chunkid 2476 (off:0, size:1, clen: 1): get chunks/0/2/2476_0_1: open /var/jfs/clickhouse/chunks/0/2/2476_0_1: no such file or directory
Apr 29 11:02:38 A06-R08-I135-39-5000418 juicefs[8812]: juicefs[8812] <WARNING>: fail to read chunkid 2476 (off:0, size:1, clen: 1): get chunks/0/2/2476_0_1: open /var/jfs/clickhouse/chunks/0/2/2476_0_1: no such file or directory
Apr 29 11:02:40 A06-R08-I135-39-5000418 juicefs[8812]: juicefs[8812] <WARNING>: fail to read chunkid 2476 (off:0, size:1, clen: 1): get chunks/0/2/2476_0_1: open /var/jfs/clickhouse/chunks/0/2/2476_0_1: no such file or directory
Apr 29 11:02:42 A06-R08-I135-39-5000418 juicefs[8812]: juicefs[8812] <WARNING>: fail to read chunkid 2476 (off:0, size:1, clen: 1): get chunks/0/2/2476_0_1: open /var/jfs/clickhouse/chunks/0/2/2476_0_1: no such file or directory
Apr 29 11:02:45 A06-R08-I135-39-5000418 juicefs[8812]: juicefs[8812] <WARNING>: fail to read chunkid 2476 (off:0, size:1, clen: 1): get chunks/0/2/2476_0_1: open /var/jfs/clickhouse/chunks/0/2/2476_0_1: no such file or directory
Apr 29 11:02:48 A06-R08-I135-39-5000418 juicefs[8812]: juicefs[8812] <WARNING>: fail to read chunkid 2476 (off:0, size:1, clen: 1): get chunks/0/2/2476_0_1: open /var/jfs/clickhouse/chunks/0/2/2476_0_1: no such file or directory
Apr 29 11:02:52 A06-R08-I135-39-5000418 juicefs[8812]: juicefs[8812] <WARNING>: fail to read chunkid 2476 (off:0, size:1, clen: 1): get chunks/0/2/2476_0_1: open /var/jfs/clickhouse/chunks/0/2/2476_0_1: no such file or directory
Apr 29 11:02:52 A06-R08-I135-39-5000418 juicefs[8812]: juicefs[8812] <ERROR>: read file 432: input/output error
Apr 29 11:02:52 A06-R08-I135-39-5000418 juicefs[8812]: juicefs[8812] <INFO>: slow operation: read (432,4096,0): input/output error (0) <20.815511>
Apr 29 11:02:53 A06-R08-I135-39-5000418 juicefs[8812]: juicefs[8812] <WARNING>: fail to read chunkid 2450 (off:0, size:78, clen: 78): get chunks/0/2/2450_0_78: open /var/jfs/clickhouse/chunks/0/2/2450_0_78: no such file or directory
Apr 29 11:02:54 A06-R08-I135-39-5000418 juicefs[8812]: juicefs[8812] <WARNING>: fail to read chunkid 2450 (off:0, size:78, clen: 78): get chunks/0/2/2450_0_78: open /var/jfs/clickhouse/chunks/0/2/2450_0_78: no such file or directory
Apr 29 11:02:55 A06-R08-I135-39-5000418 juicefs[8812]: juicefs[8812] <WARNING>: fail to read chunkid 2450 (off:0, size:78, clen: 78): get chunks/0/2/2450_0_78: open /var/jfs/clickhouse/chunks/0/2/2450_0_78: no such file or directory
Apr 29 11:02:56 A06-R08-I135-39-5000418 juicefs[8812]: juicefs[8812] <WARNING>: fail to read chunkid 2450 (off:0, size:78, clen: 78): get chunks/0/2/2450_0_78: open /var/jfs/clickhouse/chunks/0/2/2450_0_78: no such file or directory
Apr 29 11:02:58 A06-R08-I135-39-5000418 juicefs[8812]: juicefs[8812] <WARNING>: fail to read chunkid 2450 (off:0, size:78, clen: 78): get chunks/0/2/2450_0_78: open /var/jfs/clickhouse/chunks/0/2/2450_0_78: no such file or directory
Apr 29 11:03:01 A06-R08-I135-39-5000418 juicefs[8812]: juicefs[8812] <WARNING>: fail to read chunkid 2450 (off:0, size:78, clen: 78): get chunks/0/2/2450_0_78: open /var/jfs/clickhouse/chunks/0/2/2450_0_78: no such file or directory
Apr 29 11:03:03 A06-R08-I135-39-5000418 juicefs[8812]: juicefs[8812] <WARNING>: fail to read chunkid 2450 (off:0, size:78, clen: 78): get chunks/0/2/2450_0_78: open /var/jfs/clickhouse/chunks/0/2/2450_0_78: no such file or directory
Apr 29 11:03:06 A06-R08-I135-39-5000418 juicefs[8812]: juicefs[8812] <WARNING>: fail to read chunkid 2450 (off:0, size:78, clen: 78): get chunks/0/2/2450_0_78: open /var/jfs/clickhouse/chunks/0/2/2450_0_78: no such file or directory
Apr 29 11:03:09 A06-R08-I135-39-5000418 juicefs[8812]: juicefs[8812] <WARNING>: fail to read chunkid 2450 (off:0, size:78, clen: 78): get chunks/0/2/2450_0_78: open /var/jfs/clickhouse/chunks/0/2/2450_0_78: no such file or directory
Apr 29 11:03:12 A06-R08-I135-39-5000418 juicefs[8812]: juicefs[8812] <WARNING>: fail to read chunkid 2450 (off:0, size:78, clen: 78): get chunks/0/2/2450_0_78: open /var/jfs/clickhouse/chunks/0/2/2450_0_78: no such file or directory
Apr 29 11:03:12 A06-R08-I135-39-5000418 juicefs[8812]: juicefs[8812] <ERROR>: read file 443: input/output error
Apr 29 11:03:12 A06-R08-I135-39-5000418 juicefs[8812]: juicefs[8812] <INFO>: slow operation: read (443,4096,0): input/output error (0) <20.815153>

**Comment by davies** [comment]
It looks like some chunks are missing from `/var/jfs/clickhouse`, can you reproduce this with a new volume?

**Comment by davies** [comment]
What's the version of JuiceFS and when do you created this volume?

**Comment by Sallery-X** [comment]
> What's the version of JuiceFS and when do you created this volume?

0.12.1, yesterday

**Comment by Sallery-X** [comment]
two node operate one file

**Comment by Sallery-X** [comment]
I'm trying to make clickhouse to share data among replicas

**Comment by davies** [comment]
@Sallery-X You are using `local disk` as the object storage, which is only used for test purpose, because it can't be accessed from multiple machines.

In order to share JuiceFS across multiple machines you need to use other object store, for example, S3.
